elkscmd/tui/cons.c

```diff
@@ -55,9 +55,7 @@ static void display(void)
                 fputs(buf, stdout);
             }
         }
-        if (r == LINES - 1)
-            printf("\r");
-        else printf("\n");
+        putc(r == LINES - 1 ? '\r' : '\n', stdout);
```

Comment:
Similar to printf->fputs described above, doesn't this just drag in `fputc`, while also including `printf`? I would guess this results in large .o and final codefile ? (Same issue in tty-cp437.c).

Reply:
I think my initial motivation here was to save a few CPU cycles by avoiding the unnecessary format string parsing in vfprintf (which, when multiplied across screen redraws, might actually matter).

I just did a test:

Reverting cons.c and tty-cp437.c to printf as before, and doing a clean build:
```
$ du -b target/bin/
1337353	target/bin/
```

Clean build with cons.c and tty-cp437.c putc change as first presented here.  Slightly trading space for time:
```
$ du -b target/bin/
1337369	target/bin/
```

Then converting cons.c and tty-cp437.c to fputs/putc (no printf), and doing a clean build:
```
$ du -b target/bin/
1337321	target/bin/
```

So it seems that avoiding printf where possible in tty code is a [tiny] win, whether the goal is to optimize for space or time.